app/build.gradle

```
@@ -29,8 +29,8 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
```

Does this need to target Java 11? Since we're switching sourceCompatibility to Java 11 we get the Java 9 cross-compile feature, so we could technically write Java with Java 11 features but compile down to a Java 8 compatible binary
```suggestion
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_1_8
```

Essentially what that should mean is:

- We can write Java 11 code
- Consumers can continue to use a 1.8 JDK when consuming this library

---

I tested the suggestion and got this error.
```
Cause: warning: source release 11 requires target release 11
```

I think we need both to be java 11 because of the guideline on https://developer.android.com/studio/releases/gradle-plugin#java-11.

---

Ah, I just did some more searching, and it seems that cross-compile feature isn't meant to be used to backwards-compile, so `--target` cannot be lower than `--source`